tests/dist/DistTestExecutor.h

```diff
@@ -44,4 +41,15 @@ class DistTestExecutorFactory : public faabric::scheduler::ExecutorFactory
     std::shared_ptr<faabric::scheduler::Executor> createExecutor(
       faabric::Message& msg) override;
 };
+
+typedef int (*ExecutorFunction)(
```

Comment:
While a function pointer is more efficient, a `std::function<int(tests::DistTestExecutor*,...)>` could also accept lambdas with captures

Reply:
Yes good point, that would definitely be clearer. This was originally a bit hacked together but has proven quite useful and expanded. Rather than change it now and further bloat this PR I'd like to give the distributed tests a bit more love in general in another PR.